CMakeLists.txt

```diff
@@ -72,7 +73,7 @@ set(SC_TOOLS_SRC "${SC_MACHINE_ROOT}/tools")
 
 
 if (NOT ${SC_USE_SANITIZER} STREQUAL "none")
-    if (CMAKE_C_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+    if (NOT CMAKE_CXX_COMPILER_ID MATCHES ".*Clang")
```

Comment:
Please, update this line to check `C` compiler, cause it will cause problems if you use `clang++` as `C++` compiler and `gcc` as `C` compiler.

if (NOT CMAKE_CXX_COMPILER_ID MATCHES ".*Clang" OR NOT CMAKE_C_COMPILER_ID MATCHES ".*Clang")

Reply:
Please, check #47